### PR TITLE
clang-tidy 2

### PR DESCRIPTION
**Pull request by neheb**
First change is to fix deprecated usleep usage, which is optionally not available everywhere. Other changes are more clang-tidy ones.

### CODE REVIEW

File: src/fish_tests.cpp [hunk 1]
```diff
@@ -1033,10 +1034,10 @@ static void test_parser() {
 static void test_1_cancellation(const wchar_t *src) {
     auto filler = io_bufferfill_t::create(io_chain_t{});
     pthread_t thread = pthread_self();
-    double delay = 0.25 /* seconds */;
+    auto delay = static_cast<long int>(0.25 * 1E6) /* seconds */;
```
Comment (neheb) [inline src/fish_tests.cpp]: I'm not too sure of this cast. I tried using std:chrono::duration and std::chrono::microseconds to no avail.

With GCC, it reduces to long int, but maybe it reduces to long long int somewhere.

Comment (neheb) [reply src/fish_tests.cpp]: Actually yeah. On 32-bit systems.

**Comment by faho** [comment]
So sleep_for makes me nervous considering the number of crappy stdlibs out there. TBH I'd prefer keeping usleep - the devil we know.

I'm not in love with skipping empty initialization - that's really subtle behavior.

Bool literals are the right thing to use (and I'm pretty sure we have a bunch of places left where we use int when we mean bool - from the C-to-C++-conversion years ago).

Simplifying boolean expressions is good.

`using` here is just `typedef` spelt differently - I don't see the point.

Requiring `strcmp` et al to explicitly specify `!= 0` seems a bit weird but whatever.

The rest are alright.

In summary: I don't love C++.

**Comment by neheb** [comment]
usleep is deprecated by POSIX 2008. could switch to nanosleep as it recommends, but it’s uglier than sleep_for.

strcmp can return negative values AFAIK.

**Comment by ridiculousfish** [comment]
Let's take these except for the usleep one and the perf one - the usleep for the reason that faho gives and perf because, while clang-tidy spotted some mistakes, the fixes are often the wrong way and I would rather fix the important cases by hand. Thank you!

**Comment by ridiculousfish** [comment]
Merged in e4a7e7d4f730c63b27020c12b662fe74d5f46e60

**Pull request closed without merging**